### ISSUE TITLE
auto-capture encounters killed driver sessions

### ISSUE THREAD

**Issue by grawk** [paypal/nemo-screenshot#21]
After this PR:
https://github.com/paypal/nemo-screenshot/pull/15

If you use nemo-screenshot with auto-capture in an execution context where you kill and restart driver session multiply, nemo-screenshot will attempt to complete a screenshot without a session. See the below stack:

``` sh
Error: This driver instance does not have a valid session ID (did you call WebDriver.quit()?) and may no longer be used.
    at checkHasNotQuit (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:377:13)
    at webdriver.WebDriver.schedule (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:324:3)
    at webdriver.WebDriver.takeScreenshot (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:979:15)
    at Object.nemo.screenshot.snap (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo-screenshot/index.js:79:16)
    at /Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo-screenshot/index.js:141:42
    at webdriver.EventEmitter.emit (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/events.js:52:17)
    at webdriver.promise.ControlFlow.execute (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/promise.js:1352:8)
    at webdriver.WebDriver.schedule (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:345:15)
    at webdriver.WebElement.schedule_ (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:1727:23)
    at webdriver.WebElement.click (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/nemo/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:1832:15)
    at Context.<anonymous> (/Users/medelman/src/_nemo/o/nemo-example-app/test/functional/spec/view-spec.js:21:30)
    at Test.Runnable.run (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runnable.js:217:15)
    at Runner.runTest (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runner.js:373:10)
    at /Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runner.js:451:12
    at next (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runner.js:298:14)
    at /Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runner.js:308:7
    at next (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runner.js:246:23)
    at Immediate._onImmediate (/Users/medelman/src/_nemo/o/nemo-example-app/node_modules/mocha/lib/runner.js:275:5)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

@pmpranav would you be able to take a look and suggest a resolution?

**Comment by pmpranav** [comment]
@grawk Is this the nemo example app? if not can you provide the link to this repo. I can work on this after this this friday.  At the very least we can add an errback or try catch block just so that it doesn't fail the testcase.

**Comment by ibihim** [comment]
Just to make you aware, it isn't a good idea to kill webdriver sessions. If they are not quit properly they like to use a lot of trash data back. After a year it might cumulate into a couple of Gigabytes.

**Comment by grawk** [comment]
@ibihim nemo-screenshot doesn't kill webdriver sessions. But it doesn't pause the promise flow while it takes these auto-capture screenshots. Thus it can be in the process of taking a screenshot after the parent test suite has already killed the session.

**Comment by grawk** [comment]
Closed via #26
